src/dataflow/src/render/join/mod.rs

```diff
@@ -43,7 +43,7 @@ pub use linear_join::LinearJoinPlan;
 /// A complete enumeration of possible join plans to render.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum JoinPlan {
-    Linear(LinearJoinPlan),
+    Linear(Box<LinearJoinPlan>),
```

Comment:
for me, it seems clear that we emit `MaterializedEvent`s frequently, and reducing the size for most of the events is probably worth it. It seems less clear to me for this case (maybe we would be minting join plans frequently if we didn't hit the fast path on SELECTs but its non-obvious that we need to optimize that)